src/migration-scripts/ntp/2-to-3

```diff
@@ -0,0 +1,62 @@
+#!/usr/bin/env python3
+
+# Copyright (C) 2023 VyOS maintainers and contributors
+#
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License version 2 or later as
+# published by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+# T5154: allow only one ip address per family for parameter 'listen-address'
+# Allow only one interface for parameter 'interface'
+# If more than one are specified, remove such entries
+
+import sys
+
+from vyos.configtree import ConfigTree
+from vyos.template import is_ipv4
+from vyos.template import is_ipv6
+
+if (len(sys.argv) < 1):
+    print("Must specify file name!")
+    sys.exit(1)
+
+file_name = sys.argv[1]
+
+with open(file_name, 'r') as f:
+    config_file = f.read()
+
+config = ConfigTree(config_file)
+
+base_path = ['service', 'ntp']
+if not config.exists(base_path):
+    # Nothing to do
+    sys.exit(0)
+
+if config.exists(base_path + ['listen-address']) and (len([addr for addr in config.return_values(base_path + ['listen-address']) if is_ipv4(addr)]) > 1):
```

Comment:
Can these two statements doing the same `config.exists` check be nested under one check? Should make the lines more compact and readable.

```
if config.exists(base_path + ['listen-address']):
    if (len([ ... ipv4:
        ...
    if (len([ ... ipv6:
        ....
```

Also XML version needs to incremented.

Reply:
Also I assume the intention is delete all if multiple present, will chrony then listen on all ipv4 or ipv6 if no respective value is provided?

Reply:
> Can these two statements doing the same `config.exists` check be nested under one check? Should make the lines more compact and readable.

I guess you mean if statements at line 43 and line 48. It's done twice because while testing I face an error when:
 - First `if config.exists` is true, then delete all ipv4 entries.
 - If no ipv6 address us specified, then the second part of the second if (assuming the first part of the i doesn't exists), will return an error because such config doesn't exists. That's why I needed to re-check that config exists after deleting the  ipv4 addresses

Reply:
> Also I assume the intention is delete all if multiple present, will chrony then listen on all ipv4 or ipv6 if no respective value is provided?

That's correct! I no interace|address is specified, it listens on all interfaces|addresses

Example:
- Define only ipv6
-  Check that listen only in that ipv6 address and in all ipv4 addresses
```
vyos@NTP# run show config comm | grep ntp
set service ntp allow-client address '0.0.0.0/0'
set service ntp allow-client address '::/0'
set service ntp listen-address '2001:db8::1'
set service ntp server time1.vyos.net
set service ntp server time2.vyos.net
set service ntp server time3.vyos.net
[edit]
vyos@NTP# sudo netstat -putane | grep chrony
udp        0      0 0.0.0.0:123             0.0.0.0:*                           0          15614      2805/chronyd        
udp        0      0 127.0.0.1:323           0.0.0.0:*                           0          15612      2805/chronyd        
udp6       0      0 2001:db8::1:123         :::*                                0          15615      2805/chronyd        
udp6       0      0 ::1:323                 :::*                                0          15613      2805/chronyd        
[edit]
```

Reply:
Nice, just needs XML version updating: https://github.com/vyos/vyos-1x/blob/current/interface-definitions/include/version/ntp-version.xml.i

Reply:
I forgot to trigger it yesterday, my bad!